### PR TITLE
Adding BDTs, flag fixes, new branches.

### PR DESCRIPTION
**Pull request by gmadigan**
Adding in BDT classifiers and branches for single and pair production (configurable with switches). Adding in a DR variable. Customizing 2017 nominal LHE scale weight assignment.

### CODE REVIEW

File: NTupleAnalyzer_nanoAOD.py [hunk 1]
```diff
@@ -132,6 +142,7 @@
 _kinematicvariables += ['M_jj']
 _kinematicvariables += ['DR_muon1muon2','DPhi_muon1met','DPhi_jet1met','DPhi_jet2met']
 _kinematicvariables += ['DR_muon1jet1','DR_muon1jet2','DR_muon2jet1','DR_muon2jet2']
+_kinematicvariables += ['DR_muon1muon2jet1']
```
Comment (DavidMorse) [inline NTupleAnalyzer_nanoAOD.py]: I don't like this name, all the others, e.g. DR_muon1muon2 means the DR between muon1 and muon2, but this one means DR between the 2-muon object and jet1, so maybe let's call it DR_dimujet1 ? do you have a better idea?

File: NTupleAnalyzer_nanoAOD.py [hunk 2]
```diff
@@ -1875,6 +1967,7 @@ def FullKinematicCalculation(T,variation):
 	_DRu1j2 = abs(muons[0].DeltaR(jets[1]))
 	_DRu2j1 = abs(muons[1].DeltaR(jets[0]))
 	_DRu2j2 = abs(muons[1].DeltaR(jets[1]))
+	_DRu1u2j1 = abs((muons[0]+muons[1]).DeltaR(jets[0]))
```
Comment (DavidMorse) [inline NTupleAnalyzer_nanoAOD.py]: same comment here, maybe call this _DRdimujet0 ? Do you have a better idea?

File: NTupleAnalyzer_nanoAOD.py [hunk 3]
```diff
@@ -1971,6 +2079,7 @@ def FullKinematicCalculation(T,variation):
 	toreturn += [_Mjj]
 	toreturn += [_DRuu,_DPHIuv,_DPHIj1v,_DPHIj2v]
 	toreturn += [_DRu1j1,_DRu1j2,_DRu2j1,_DRu2j2]
+	toreturn += [_DRu1u2j1]
```
Comment (DavidMorse) [inline NTupleAnalyzer_nanoAOD.py]: same comment here

**Review by DavidMorse — CHANGES_REQUESTED** [review]

File: NTupleAnalyzer_nanoAOD.py (hunk 1 above)
Comment (gmadigan) [reply NTupleAnalyzer_nanoAOD.py]: Yeah I agree with you, there. I went with DR_dimuonjet1 for the branch name, and _DRdimujet1 for the local variable.

**Pull request merged**